modules/gapi/test/common/gapi_video_tests_common.hpp

```diff
@@ -321,6 +322,53 @@ inline GComputation runOCVnGAPIOptFlowPipeline(TestFunctional& testInst,
     return c;
 }
 
+inline void compareBackSubResults(const cv::Mat &actual, const cv::Mat &expected, const int diffPercent)
+{
+    GAPI_Assert(actual.size() == expected.size());
+    int allowedNumDiffPixels = actual.size().area() * diffPercent / 100;
+
+    cv::Mat diff;
+    cv::absdiff(actual, expected, diff);
+
+    cv::Mat hist(256, 1, CV_32FC1, cv::Scalar(0));
+    const float range[]{ 0, 256 };
+    const float *histRange{ range };
+    calcHist(&diff, 1, 0, Mat(), hist, 1, &hist.rows, &histRange, true, false);
+    for (int i = 2; i < hist.rows; ++i)
+    {
+        hist.at<float>(i) += hist.at<float>(i - 1);
```

Comment:
Can't it be done by function like `cv::sum`?

Reply:
You can consider using `cv::norm` to compare two Mats as well

And also you can try one of comparison functions implemented in `gapi_tests_common.hpp`

Reply:
The snippet isn't my. I only moved it.

Reply:
I'm planning revert moving.

Reply:
Got it. It looks OK if you want to preserve this way of comparison
However, there are a lot of different comparison functions for `Mat`s in our `gapi_tests_common.hpp` so I suggest you to try one of them instead of `compareBackSubResults`
For example, `AbsSimilarPoints()` could probably fit

Reply:
What do you think on it then?

Reply:
Ok. Changed.